### PR TITLE
Allow creating form tag without "action" attribute.

### LINKED ISSUE #6488: 3.x FormHelper no way to create form without action
There doesn't seem to be a way to create an actionless form short of not using FormHelper.

### PR DESCRIPTION
**Pull request by ADmad**
Closes #6488

### CODE REVIEW

File: tests/TestCase/View/Helper/FormHelperTest.php [hunk 1]
```diff
@@ -674,6 +674,26 @@ public function testCreateAutoUrl()
     }
 
     /**
+     * test create() with no url (no "action" attribute for <form> tag)
```
Comment (dereuromark) [inline tests/TestCase/View/Helper/FormHelperTest.php]: nitpick: URL

Comment (jadb) [inline tests/TestCase/View/Helper/FormHelperTest.php]: nitpick on nitpick: you missed mentioning the missing capital T (beginning of sentence).

File: src/View/Helper/FormHelper.php [hunk 2]
```diff
@@ -367,11 +367,16 @@ public function create($model = null, array $options = [])
         }
         unset($options['templates']);
 
-        $url = $this->_formUrl($context, $options);
-        $action = $this->Url->build($url);
-        unset($options['url'], $options['action'], $options['idPrefix']);
+        if ($options['url'] === false) {
+            $action = null;
+            $this->_lastAction = '';
```
Comment (markstory) [inline src/View/Helper/FormHelper.php]: Forms with no action attribute post to the current URL. Blanking this property will result in SecurityComponent hashes not having the URL and subsequently failing validation.

Comment (ADmad) [inline src/View/Helper/FormHelper.php]: Yes I am aware of that and [stated](https://github.com/cakephp/cakephp/issues/6488#issuecomment-99184625) the same on related issue. Please read the discussion on #6488 for why people want this feature and why it won't be used with SecurityComponent.

We can throw an exception if `url` is set to false and `$this->request['_Token']` is not empty.

Comment (markstory) [inline src/View/Helper/FormHelper.php]: Why couldn't we just set _lastAction to the current URL?

Comment (dereuromark) [inline src/View/Helper/FormHelper.php]: How would that avoid the inclusion of the action attribute?

**Comment by dereuromark** [comment]
:+1:

File: src/View/Helper/FormHelper.php (hunk 2 above)
Comment (markstory) [inline src/View/Helper/FormHelper.php]: They are separate things.

Comment (ADmad) [inline src/View/Helper/FormHelper.php]: > Why couldn't we just set _lastAction to the current URL?

Makes sense. Updated accordingly.

**Pull request merged**